src/Agent.Worker/StepsRunner.cs

```diff
@@ -50,6 +52,21 @@ public async Task RunAsync(IExecutionContext jobContext, IList<IStep> steps)
             CancellationTokenRegistration? jobCancelRegister = null;
             int stepIndex = 0;
             jobContext.Variables.Agent_JobStatus = jobContext.Result ?? TaskResult.Succeeded;
+            // Wait till all async commands finish.
+            foreach (var command in jobContext.AsyncCommands ?? new List<IAsyncCommandContext>())
```

Comment:
[Question] Are we sure we want to wait all async commands here?
Probably, should we wait only specific commands?

Reply:
actually on my first approach I added boolean parameter to define which commands we should wait. but as there is not another commands except current one, I removed this parameter.